### PR TITLE
Hub admins can now manage Kubernetes namespaces, assigned to project services

### PR DESCRIPTION
**Pull request by jits**
Note: there's currently no relationship/interaction between RBAC groups and namespaces within the hub – we can solve this in a later PR (depending on _if_ and _how_ we might want to do that).

### CODE REVIEW

File: platform-hub-api/app/models/kubernetes_namespace.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  validates :name,
+    format: {
+      with: NAME_REGEX,
+      message: "should consist of lowercase letters, numbers and dashes, and must start and end with an alphanumeric character"
+    },
+    uniqueness: {
+      scope: :cluster_id,
+      message: "already exists for the cluster"
+    }
```
Comment (marcinc) [inline platform-hub-api/app/models/kubernetes_namespace.rb]: Should we also validate description presence?

File: platform-hub-web/src/app/kubernetes/kubernetes-namespaces-form.component.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    ctrl.loading = true;
+
+    ctrl.isNew = !namespaceId;
+    ctrl.namespace = null;
+
+    if (
+      (!fromProject && fromService) ||
+      (fromProject && !fromService)
+    ) {
+      logger.error('Bug detected: both `fromProject` and `fromService` need to be provided, or neither');
```
Comment (marcinc) [inline platform-hub-web/src/app/kubernetes/kubernetes-namespaces-form.component.js]: nice error message :) 🐛

File: platform-hub-web/src/app/kubernetes/kubernetes-namespaces-form.html [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+              required
+              ng-pattern="/^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/">
+            <div ng-messages="$ctrl.kubernetesNamespaceForm.name.$error">
+              <div ng-message="required">This is required.</div>
+              <div ng-message="pattern">Only lowercase letters, numbers and dashes, and must start and end with an alphanumeric character.</div>
+            </div>
+          </md-input-container>
+
+          <md-input-container class="md-block">
+            <label for="description">Description</label>
```
Comment (marcinc) [inline platform-hub-web/src/app/kubernetes/kubernetes-namespaces-form.html]: `required` if we decide this should be always present.

File: platform-hub-web/src/app/kubernetes/kubernetes-namespaces-form.component.js (hunk 2 above)
Comment (jits) [reply platform-hub-web/src/app/kubernetes/kubernetes-namespaces-form.component.js]: 👍  bit defensive coding here; it should never happen… but you never know ;-)

File: platform-hub-api/app/models/kubernetes_namespace.rb (hunk 1 above)
Comment (jits) [reply platform-hub-api/app/models/kubernetes_namespace.rb]: I was thinking we make description optional for namespaces. Seems like the namespace name alone is often descriptive enough, so requiring a description would just make people put the same name as the description.

**Pull request merged**